lambda/transl_comprehension_utils.ml

```diff
@@ -0,0 +1,167 @@
+open Lambda
+
+module Let_binding = struct
+  module Let_kind = struct
+    type t =
+      | Immutable of let_kind
+      | Mutable
+  end
+
+  type t =
+    { let_kind   : Let_kind.t
+    ; value_kind : value_kind
+    ; id         : Ident.t
+    ; init       : lambda }
+
+  let make_id let_kind value_kind name init =
+    let id = Ident.create_local name in
+    id, {let_kind; value_kind; id; init}
+
+  let make_id' let_kind value_kind name init =
+    snd @@ make_id let_kind value_kind name init
+
+  let make_id_var let_kind value_kind name init =
+    let id, binding = make_id let_kind value_kind name init in
+    id, Lvar id, binding
+
+  let make_var let_kind value_kind name init =
+    let _id, var, binding = make_id_var let_kind value_kind name init in
+    var, binding
+
+  let let_one {let_kind; value_kind; id; init} body =
+    match let_kind with
+    | Immutable let_kind -> Llet(let_kind, value_kind, id, init, body)
+    | Mutable            -> Lmutlet(value_kind, id, init, body)
+
+  let let_all = List.fold_right let_one
+end
+
+module Lambda_utils = struct
+  module Make = struct
+    let int n = Lconst (const_int n)
+
+    let float f = Lconst (Const_base (Const_float (Float.to_string f)))
+
+    let string ~loc s = Lconst (Const_base (Const_string(s, loc, None)))
+  end
+
+  (* CR aspectorzabusky: I think [alloc_heap] is the right default for
+     [ap_mode], but I don't understand what [ap_mode] means. *)
+  let apply
```

Comment:
I tend to favor adding these kinds of features lazily -- only when they're needed. When actual uses come up, there is often clarifying context, and the original formulation of the idea might be wrong, anyway.

Reply:
The difference here is that `apply` is really just a wrapper around `Lapply`, so there's not exactly any choice in what the optimal arguments are.  Still, perhaps that's a case for writing out `Lapply` if you want things, since I am picking defaults.  I'll make the change.